src/common/main/kotlin/tech/thatgravyboat/skyblockapi/api/profile/currency/CurrencyAPI.kt

```diff
@@ -1,27 +1,37 @@
 package tech.thatgravyboat.skyblockapi.api.profile.currency
 
 import me.owdding.ktmodules.Module
+import tech.thatgravyboat.skyblockapi.api.data.stored.CommunityCenterStorage
 import tech.thatgravyboat.skyblockapi.api.data.stored.CurrencyStorage
 import tech.thatgravyboat.skyblockapi.api.events.base.Subscription
 import tech.thatgravyboat.skyblockapi.api.events.base.predicates.OnlyOnSkyBlock
 import tech.thatgravyboat.skyblockapi.api.events.base.predicates.OnlyWidget
+import tech.thatgravyboat.skyblockapi.api.events.info.CurrencyUpdateEvent
 import tech.thatgravyboat.skyblockapi.api.events.info.ScoreboardUpdateEvent
 import tech.thatgravyboat.skyblockapi.api.events.info.TabWidget
 import tech.thatgravyboat.skyblockapi.api.events.info.TabWidgetChangeEvent
 import tech.thatgravyboat.skyblockapi.api.location.SkyBlockIsland
-import tech.thatgravyboat.skyblockapi.api.profile.community.CommunityCenterAPI
 import tech.thatgravyboat.skyblockapi.utils.extentions.parseFormattedDouble
 import tech.thatgravyboat.skyblockapi.utils.extentions.parseFormattedLong
 import tech.thatgravyboat.skyblockapi.utils.regex.RegexGroup
 import tech.thatgravyboat.skyblockapi.utils.regex.RegexUtils.anyFound
 import tech.thatgravyboat.skyblockapi.utils.regex.RegexUtils.anyMatch
 
-enum class PurseType {
-    UNKNOWN,
-    NORMAL,
+enum class PurseType(scoreboardName: String? = null) {
+    NORMAL("PURSE"),
     PIGGY,
+    UNKNOWN,
+    ;
+
+    private val scoreboardName: String = scoreboardName ?: name
+
+    companion object {
+        fun fromName(name: String): PurseType = entries.find { it.scoreboardName.equals(name, true) } ?: UNKNOWN
+    }
 }
 
+private typealias CurrencyEvent = CurrencyUpdateEvent<*>
```

Comment:
the typealias is only used for the scoreboard events not the tab events it should be used consistently